### PR TITLE
Allow cancelling operations with SIGTERM

### PR DESCRIPTION
**Pull request by julienp**
We already handle SIGINT to cancel operations. This PR additionally
handles SIGTERM the same way. This is useful when the Pulumi CL is used
in a docker container, where a stop command sends a SIGTERM. Previously
this would immediately terminate the CLI.

Fixes https://github.com/pulumi/pulumi/issues/17256

### CODE REVIEW

File: pkg/backend/cancellation_scope.go [hunk 1]
```diff
@@ -49,15 +50,15 @@ func (cancellationScopeSource) NewScope(events chan<- engine.Event, isPreview bo
 
 	c := &cancellationScope{
 		context: cancelContext,
-		sigint:  make(chan os.Signal),
+		sigint:  make(chan os.Signal, 1),
```
Comment (julienp) [inline pkg/backend/cancellation_scope.go:53]: The channel for `signal.Notify` should be buffered https://pkg.go.dev/os/signal#Notify

File: pkg/backend/cancellation_scope.go [hunk 2]
```diff
@@ -49,16 +50,19 @@ func (cancellationScopeSource) NewScope(events chan<- engine.Event, isPreview bo
...
-		for range c.sigint {
-			// If we haven't yet received a SIGINT, call the cancellation func. Otherwise call the termination
-			// func.
+		for sig := range c.sigint {
+			// If we haven't yet received a SIGINT or SIGTERM, call the cancellation func. Otherwise call the
+			// termination func.
 			if cancelContext.CancelErr() == nil {
 				message := "^C received; cancelling. If you would like to terminate immediately, press ^C again.\n"
+				if sig == syscall.SIGTERM {
+					message = "SIGTERM received; cancelling. If you would like to terminate immediately, send SIGTERM again.\n"
```
Comment (julienp) [inline pkg/backend/cancellation_scope.go:64]: I updated the message here, but I remember I looked into system messages (`engine.NewEvent(engine.StdoutEventPayload{...})`) at some time before, and I never managed to get any to display. I am not sure they ever show up I the current UI.

File: pkg/backend/cancellation_scope.go (hunk 1 above)
Comment (lunaris) [reply pkg/backend/cancellation_scope.go:53]: Worth adding this as a comment to the code? Seems like a useful thing to call out.

**Review by lunaris — APPROVED** [review]

File: pkg/backend/cancellation_scope.go (hunk 1 above)
Comment (julienp) [reply pkg/backend/cancellation_scope.go]: Yeah, good idea

**Pull request merged**